addons/mail/static/src/utils/test_utils.js

```diff
@@ -418,12 +418,34 @@ function getClick({ afterNextRender }) {
     };
 }
 
-function getCreateChatterContainerComponent({ components, env, widget }) {
-    return async function createChatterContainerComponent(props) {
-        return await createRootMessagingComponent({ components, env }, "ChatterContainer", {
-            props,
-            target: widget.el,
-        });
+function getCreateChatterContainerComponent({ afterEvent, components, env, widget }) {
+    return async function createChatterContainerComponent(props, { waitUntilMessagesLoaded = true } = {}) {
+        let chatterContainerComponent;
+        async function func() {
+            chatterContainerComponent = await createRootMessagingComponent({ components, env }, "ChatterContainer", {
+                props,
+                target: widget.el,
+            });
+        }
+        if (waitUntilMessagesLoaded) {
+            await afterNextRender(() => afterEvent({
+                eventName: 'o-thread-view-hint-processed',
+                func,
+                message: "should wait until chatter loaded messages after creating chatter container component",
+                predicate: ({ hint, threadViewer }) => {
+                    return (
+                        hint.type === 'messages-loaded' &&
+                        threadViewer &&
+                        threadViewer.chatter &&
+                        threadViewer.chatter.threadModel === props.threadModel &&
+                        threadViewer.chatter.threadId === props.threadId
+                    );
+                },
+            }));
+        } else {
+            await func();
+        }
+        return chatterContainerComponent;
```

Comment:
Won't it reach the branching `waitUntilmessagesLoaded` by default even when the `Chatter/hasMessageList` is `false`? I don't think we have such tests, but handling this case could prevent wasting time and being annoyed that time was wasted just because this code made shortcut.

At least the assert comment should help a bit.

Reply:
Are you asking that I condition the default depending on the chatter props? That would make sense, but I guess we can worry about it if we have the need some day instead of preemptively adding extra code. (Especially because those helpers are not meant to say here long time in their current shape)

Reply:
Ok, just wanted to point out limitation in this code in case we start making chatter tests with `Chatter/hasMessageList: false`, which I think is unlikely but who knows.